stub/storage.go

```diff
@@ -160,45 +162,96 @@ func renderFieldAsString(fields map[string]interface{}) string {
 	return template
 }
 
-func equals(input1, input2 map[string]interface{}) bool {
-	return reflect.DeepEqual(input1, input2)
+func deepEqual(expect, actual interface{}) bool {
+	return reflect.DeepEqual(expect, actual)
+}
+
+func regexMatch(expect, actual interface{}) bool {
+	match, err := regexp.Match(expect.(string), []byte(actual.(string)))
+	if err != nil {
+		log.Printf("Error on matching regex %s with %s error:%v\n", expect, actual, err)
+	}
+	return match
+}
```

Comment:
Created two functions for deep equality and regex equality. We will pass this to above defined _matchFunc_